src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -223,7 +223,12 @@ private bool ShouldConsiderSymbol(ISymbol symbol)
                 list.Add(new DocumentHighlights(kvp.Key, spans.ToImmutableAndFree()));
             }
 
-            return list.ToImmutableAndFree();
+            var result = list.ToImmutableAndFree();
+
+            // Diagnosing flaky tests: assert that no span appears more than once
+            var groupedSpans = result.SelectMany(d => d.HighlightSpans).GroupBy(s => s);
+            Contract.ThrowIfTrue(groupedSpans.Any(s => s.Count() > 1), "Found duplicate spans!");
```

Comment:
Do not check this in :)  

If you want to assert at the test level, that's fine.  But this risks breaking things by adding a new contract that wasn't ever stated before.